src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php

```diff
@@ -113,19 +121,23 @@ public function applyToItem(QueryBuilder $queryBuilder, QueryNameGeneratorInterf
      * @param bool                        $forceEager
      * @param string                      $parentAlias
      * @param array                       $propertyMetadataOptions
+     * @param array                       $context
      * @param bool                        $wasLeftJoin             if the relation containing the new one had a left join, we have to force the new one to left join too
      * @param int                         $joinCount               the number of joins
+     * @param int                         $currentDepth            the current max depth
      *
      * @throws RuntimeException when the max number of joins has been reached
      */
-    private function joinRelations(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, bool $forceEager, bool $fetchPartial, string $parentAlias, array $propertyMetadataOptions = [], bool $wasLeftJoin = false, int &$joinCount = 0)
+    private function joinRelations(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, bool $forceEager, bool $fetchPartial, string $parentAlias, array $propertyMetadataOptions = [], array $context = [], bool $wasLeftJoin = false, int &$joinCount = 0, int $currentDepth = null)
     {
         if ($joinCount > $this->maxJoins) {
-            throw new RuntimeException('The total number of joined relations has exceeded the specified maximum. Raise the limit if necessary.');
+            throw new RuntimeException('The total number of joined relations has exceeded the specified maximum. Raise the limit if necessary, or implement @MaxDepth.');
```

Comment:
`or use the "max_depth" option of the Symfony serializer`? (Max depth can be configured using a YAML or a XML config file too).